src/AnalysisPrograms/Production/MainEntryUtilities.cs

```diff
@@ -279,7 +279,11 @@ internal static void HangBeforeExit()
 
             // if Michael is debugging with visual studio, this will prevent the window closing.
             Process parentProcess = ProcessExtensions.ParentProcessUtilities.GetParentProcess();
-            if (parentProcess.ProcessName == "devenv")
+            if (parentProcess == null)
+            {
+                LoggedConsole.WriteWarnLine("WARNING: Unable to detect parent process, this is a windows specific tool.");
+            }
```

Comment:
I don't think we want to emit a warning here. The default case is that no parent process is attached. 

Maybe emit the warning only if `Debugger.IsAttached()`?